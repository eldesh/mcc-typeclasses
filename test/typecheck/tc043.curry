-- This example checks that the left hand and right hand side contexts in
-- a data type declaration are managed properly by the compiler even in
-- cases where the type constraints overlap.

import A

incr (C1 x) = C1 (x + 1)
tautology (C2 x) = x == x
