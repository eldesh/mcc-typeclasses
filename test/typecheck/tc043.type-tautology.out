Num _1 => T2 _1 -> Bool
